### DISCUSSION TITLE
PCAN Explorer files

### DISCUSSION THREAD

**Discussion by damienmaguire** [collin80/SavvyCAN#333] [Q&A]
Is it possible to read PCAN explorer .trc files in Savvycan?

**Comment by collin80** [comment]
I'm pretty sure it does. I have an entry for "PCAN Viewer" that loads .trc files. So, it seems it sure ought to.

**Reply by damienmaguire** [reply]
Yes I saw that but it just crashes when file is opened. Both Linux and windoze.
file here :https://github.com/damienmaguire/LeafLogs/tree/master/Gen3_2018

**Reply by collin80** [reply]
Well what a surly mongrel, that SavvyCAN. I'll set 'em straight.

**Reply by collin80** [reply]
It turns out that I supported version 1.1 and 2.0 files from PCAN. Those files were version 1.3. This version is now supported as well. At this rate it's almost to the point where I should parse their header to figure out the layout automatically. But, 3 versions of the file isn't to that point... yet. I've pushed the code to support this file format and the dutiful elves at Appveyor and TravisCI should be compiling the new version as I type this.

**Comment by damienmaguire** [comment]
Thanks Colin. So it now opens but lists the ids incorrectly.

**Comment by damienmaguire** [comment]
On closer inspection it seems the ids are one byte right shifted. e.g. 0x1d4 comes in as 0x01d.

**Reply by collin80** [reply]
Yeah, quality control, it's our 14th most important job! Sure enough, there was an off by one error in the code so it would fail to load the last character for the ID. This has been corrected and the person responsible has been slapped.

**Comment by damienmaguire** [comment]
Perfect now. Thanks Colin:)
